### DISCUSSION TITLE
Terraform Provider for Oxide

### DISCUSSION THREAD

**Discussion by zacksiri** [oxidecomputer/omicron#3074] [Ideas]
I was wondering if there is a plan for a terraform provider for oxide's control plane. We build continuous delivery system and use terraform to orchestrate bootstrapping clusters that get added to our platform to handle deployments. Having a terraform provider would be great for us to create a terraform module around. Here are examples of what we're working on.

https://github.com/upmaru/instellar-terraform-modules

Our goal is to add google and azure and we also want to enable bare metal cloud like Oxide on our platform.

**Comment by davepacheco** [comment]
You might be interested in https://github.com/oxidecomputer/terraform-provider-oxide?

**Reply by zacksiri** [reply]
Yeah, I've been looking for it. It's not public though, so nothing turned up when I looked.

**Reply by davepacheco** [reply]
Ugh -- sorry.  I'll look into the plan here.

**Comment by karencfv** [comment]
Hey @zacksiri, thanks for taking us into account! We're currently finishing up a few things and will make the Terraform provider public within the next month or so.

**Reply by zacksiri** [reply]
Hi @karencfv 

Thank you for the reply. I was also wondering if the oxide team would be able to make a mock api of some kind available to test against. Given that oxide rack will be difficult for small teams like us to obtain.

**Reply by davepacheco** [reply]
@zacksiri would this work for you: https://github.com/oxidecomputer/omicron/blob/main/docs/how-to-run-simulated.adoc

**Reply by zacksiri** [reply]
@davepacheco that looks good, yeah I think I can use that! Thank you!
